src/main.rs

```diff
@@ -140,20 +174,38 @@ async fn handle_inline_query(bot: Bot, query: InlineQuery) {
     bot.send(&response).await.unwrap();
 }
 
+#[inline(always)]
 fn get_char_names(string: &str) -> String {
+    _get_char_names_filtered(string, None)
+}
+
+#[inline(always)]
+fn get_char_names_filtered(string: &str, filter: &str) -> String {
+    _get_char_names_filtered(string, Some(filter))
+}
+
+fn _get_char_names_filtered(string: &str, filter: Option<&str>) -> String {
     let mut text = String::with_capacity(4096); // max telegram message length
+    let mut entities = 0;
+
+    for c in string.chars() {
+        if let Some(f) = filter {
+            if f.contains(c) {
+                continue;
+            }
+        }
 
-    for (i, c) in string.chars().enumerate() {
         let name = charname::get_name(c as u32);
 
         let new_part = format!(
             "`U+{val:04X}` [{}](https://fileformat.info/info/unicode/char/{val:X})\n",
             name,
             val = c as u32
         );
+        entities += 1;
 
         // Don't want to exceed message limit or message entity limit
-        if text.len() + new_part.len() >= 4000 || i >= 50 {
+        if text.len() + new_part.len() >= 4000 || entities >= 50 {
```

Comment:
The behavior here has changed. If i was >= 50, then that meant that we had added 50 entities already. We can have a maximum of 50 entities, so this should really be `>` instead of `>=`

```suggestion
        if text.len() + new_part.len() >= 4000 || entities > 50 {
```